src/liboslexec/backendllvm.h

```diff
@@ -150,6 +150,15 @@ class BackendLLVM : public OSOProcessorBase {
     /// reside in a global variable, the groupdata struct, or a local value.
     llvm::Value *llvm_load_device_string (const Symbol& sym);
 
+    /// Convenience function to load a string for CPU or GPU device
+    llvm::Value *llvm_load_string (const Symbol& sym) {
```

Comment:
I feel like this would be better as:

    DASSERT(sym.typespec().is_string());
    return use_optix() ? llvm_load_device_string(sym) : llvm_load_value(sym);